pkg/pubsub/adapter/adapter.go

```diff
@@ -128,30 +144,29 @@ func (a *Adapter) Start(ctx context.Context) error {
 }
 
 func (a *Adapter) receive(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
-	ctx, r := observability.NewReporter(ctx, CodecObserved{o: reportReceive})
-	err := a.obsReceive(ctx, event, resp)
-	if err != nil {
-		r.Error()
-	} else {
-		r.OK()
-	}
-	return err
-}
-
-func (a *Adapter) obsReceive(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
 	logger := logging.FromContext(ctx).With(zap.Any("event.id", event.ID()), zap.Any("sink", a.Sink))
 
+	args := &metrics.ReportArgs{
+		Name:          a.Name,
+		Namespace:     a.Namespace,
+		EventType:     event.Type(),
```

Comment:
I bet this will not be upstreamed into pkg, it assumed cloudevents...

Reply:
yeah, you are probably right... But you will be surprised the things that have been upstreamed to pkg... Metrics in particular is something that was pretty coupled with serving. It is now also coupled with eventing :), and needs a proper refactor... See this [issue](https://github.com/knative/pkg/issues/608)...

As I saw the [source_types](https://github.com/knative/pkg/blob/528ad1c1dd627059b95aef17ccf27f9ab0f46c10/apis/duck/v1/source_types.go#L59) duck type there, which has a CloudEventOverrides field, I also moved this one down to pkg in order to avoid duplicating code all over... 
If there are objections of moving this down, we can further discuss and see the best quickest approach